src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -683,19 +696,19 @@ protected virtual void ValidateState(OpenIdConnectProtocolValidationContext vali
             }
 
             // if state is missing, but not required just return. Otherwise process it.
-            if (!RequireStateValidation && string.IsNullOrEmpty(validationContext.State) && string.IsNullOrEmpty(validationContext.ProtocolMessage.State))
+            if (!RequireState && string.IsNullOrEmpty(validationContext.State) && string.IsNullOrEmpty(validationContext.ProtocolMessage.State))
```

Comment:
'Not Required' implies the value is optional rather than that it is required not to be present. What about `ExpectState` vs `!ExpectState`? Or `RequireEmptyState` vs `!RequireEmptyState`? Hmm, I don't know if there's a clear way to say this with a Boolean. You could always make it an enum: RequireState vs RequireNoState.

Regardless, don't block on my feedback for this property, I won't be using it.